autoload/fern/internal/drawer/hover_popup.vim

```diff
@@ -0,0 +1,46 @@
+function! fern#internal#drawer#hover_popup#init() abort
+  if g:fern#disable_drawer_hover_popup
+    return
+  endif
+
+  if !exists('*popup_create')
+    return
+  endif
+
+  augroup fern_internal_drawer_hover_popup_init
+    autocmd! * <buffer>
+    autocmd CursorMoved <buffer> call s:cursor_moved_event()
+  augroup END
+endfunction
+
+function! s:cursor_moved_event() abort
+  let l:helper = fern#helper#new()
+
+  if s:should_display_popup()
+    call s:show_popup(helper)
+  endif
+endfunction
+
+function! s:show_popup(helper)
+  let l:node = a:helper.sync.get_cursor_node()
+  if l:node is# v:null
+    return
+  endif
+
+  " calculate the screen column of node label
+  let l:line = getline('.')
+  let l:labelbegin = charidx(l:line, strridx(l:line, l:node.label))
+  let l:labelbegin = l:labelbegin < 0 ? 0 : l:labelbegin
+  let l:col_offset = wincol() - getcursorcharpos()[2] + l:labelbegin
```

Comment:
I'm kinda struggling on writing these tests. This might take me a bit.

Reply:
OK, then push your changes without tests.

Reply:
I think I'm starting to get some traction now :-) Thanks for being patient!

Reply:
Alright, I've got some tests written but they are currently failing and I'm not sure where to go from here. Hoping you might be able to give some guidance.

I'm not too familiar with `vim-themis`, but I gather that the tests are executed in a "headless" sort of environment. `calculate_node_char_offset()` uses `screenpos()` to calculate the offsets correctly, but because the tests don't actually run with a full-featured tty, it returns all zeroes and doesn't produce the correct values in test.

I could hack around it by accepting the result from `screenpos()` as an argument but that feels icky. How would you approach this problem?